rllib/algorithms/dreamerv3/utils/env_runner.py

```diff
@@ -0,0 +1,535 @@
+"""
+[1] Mastering Diverse Domains through World Models - 2023
+D. Hafner, J. Pasukonis, J. Ba, T. Lillicrap
+https://arxiv.org/pdf/2301.04104v1.pdf
+
+[2] Mastering Atari with Discrete World Models - 2021
+D. Hafner, T. Lillicrap, M. Norouzi, J. Ba
+https://arxiv.org/pdf/2010.02193.pdf
+"""
+from collections import defaultdict
+from functools import partial
+from typing import List, Tuple
+
+import gymnasium as gym
+import numpy as np
+from supersuit.generic_wrappers import resize_v1
+import tensorflow as tf
+import tree  # pip install dm_tree
+
+from ray.rllib.algorithms.algorithm_config import AlgorithmConfig
+from ray.rllib.core.models.base import STATE_IN, STATE_OUT
+from ray.rllib.env.env_runner import EnvRunner
+from ray.rllib.env.wrappers.atari_wrappers import NoopResetEnv, MaxAndSkipEnv
+from ray.rllib.env.wrappers.dm_control_wrapper import DMCEnv
+from ray.rllib.evaluation.metrics import RolloutMetrics
+from ray.rllib.policy.sample_batch import DEFAULT_POLICY_ID, SampleBatch
+from ray.rllib.utils.annotations import override
+from ray.rllib.utils.replay_buffers.episode_replay_buffer import _Episode as Episode
+from ray.rllib.utils.numpy import one_hot
+
+
+class DreamerV3EnvRunner(EnvRunner):
+    """An environment runner to collect data from vectorized gymnasium environments."""
+
+    def __init__(
+        self,
+        config: AlgorithmConfig,
+        **kwargs,
+    ):
+        """Initializes a DreamerV3EnvRunner instance.
+
+        Args:
+            config: The config to use to setup this EnvRunner.
+        """
+        super().__init__(config=config)
+
+        # Create the gym.vector.Env object.
+        if self.config.env.startswith("ALE"):
```

Comment:
I think, on the new gymnasium we won't need this. (I say this because of this recent PR that I reviewed from rohan). https://github.com/ray-project/ray/pull/35698

Reply:
Fixed this in the AlgorithmConfig API:
* Changed AlgorithmConfig.is_atari into a property (instead of setting this in the `validate()` call, which we should not with any properties)
* We can then just "outsource" this check here and replace the above if block with:
```
if self.confg.is_atari:
```